src/models/HydroReservoir.jl

```diff
@@ -0,0 +1,128 @@
+    name::String
+    "Indicator of whether the component is connected and online (`true`) or disconnected, offline, or down (`false`). Unavailable components are excluded during simulations"
+    available::Bool
+    "Volume limits for the reservoir in m^3. If nothing, the reservoir volume is assumed to be infinite."
+    storage_volume_limits::Union{Nothing, MinMax, TimeSeriesKey}
+    "Amount of water allowed to be spilled from the reservoir. If nothing, no spillage is allowed."
+    spillate_outflow_limits::Union{Nothing, MinMax, TimeSeriesKey}
+    "Amount of water refilling the reservoir in m^3/h."
+    inflow::Union{Float64, TimeSeriesKey}
+    "Amount of water going to the turbine(s) in m^3/h."
+    outflow::Union{Float64, MinMax, TimeSeriesKey}
+    "Volume target at the end of a simulation as a fraction of the total volume"
+    volume_targets::Union{Float64, MinMax, TimeSeriesKey}
+    "Downstream travel time in hours"
+    travel_time::Union{Float64, MinMax, TimeSeriesKey}
+    "Services that this device contributes to"
+    services::Vector{Service}
+    "An [*ext*ra dictionary](@ref additional_fields) for users to add metadata that are not used in simulation, such as latitude and longitude."
+    ext::Dict{String, Any}
+    "(**Do not modify.**) PowerSystems.jl internal reference"
+    internal::InfrastructureSystemsInternal
+end
+
+function HydroReservoir(name, available, storage_volume_limits, spillate_outflow_limits, inflow, outflow, volume_targets, travel_time, services=Device[], ext=Dict{String, Any}(), )
+    HydroReservoir(name, available, storage_volume_limits, spillate_outflow_limits, inflow, outflow, volume_targets, travel_time, services, ext, InfrastructureSystemsInternal(), )
+end
+
+function HydroReservoir(; name, available, storage_volume_limits, spillate_outflow_limits, inflow, outflow, volume_targets, travel_time, services=Device[], ext=Dict{String, Any}(), internal=InfrastructureSystemsInternal(), )
+    HydroReservoir(name, available, storage_volume_limits, spillate_outflow_limits, inflow, outflow, volume_targets, travel_time, services, ext, internal, )
+end
+
+# Constructor for demo purposes; non-functional.
+function HydroReservoir(::Nothing)
+    HydroReservoir(;
+        name="init",
+        available=false,
+        storage_volume_limits=nothing,
+        spillate_outflow_limits=nothing,
+        inflow=0.0,
+        outflow=0.0,
+        volume_targets=0.0,
+        travel_time=0.0,
+        services=Device[],
+        ext=Dict{String, Any}(),
+    )
+end
+
+"""Get [`HydroReservoir`](@ref) `name`."""
+get_name(value::HydroReservoir) = value.name
+"""Get [`HydroReservoir`](@ref) `available`."""
+get_available(value::HydroReservoir) = value.available
+"""Get [`HydroReservoir`](@ref) `storage_volume_limits`."""
+get_storage_volume_limits(value::HydroReservoir) = value.storage_volume_limits
```

Comment:
These getter and setter methods are not being exported, as they are not automatically included in the includes.jl files.